src/Sarif/Schemata/sarif-schema.json

```diff
@@ -124,12 +124,14 @@
 
         "offset": {
           "description": "The offset in bytes of the artifact within its containing artifact.",
-          "type": "integer"
+          "type": "integer",
+          "minimum": 0
         },
 
         "length": {
           "description": "The length of the artifact in bytes.",
-          "type": "integer"
+          "type": "integer",
+          "minimum": 0
```

Comment:
>minimum [](start = 11, length = 7)

Is this right? How do we differentiate a zero-byte length artifact from one that hasn't had a length specified? Note that this note doesn't apply to 'offset', we can assume that 0 is correct in the default case (i.e., that the artifact isn't offset from a parent). @lgolding

Reply:
To be clear, I wonder whether there shouldn't be a default value of -1 here indicating (not specified). With a min of -1.

---
In reply to: [276421274](https://github.com/Microsoft/sarif-sdk/pull/1428#discussion_r276421274) [](ancestors = 276421274)

Reply:
spec text for reference: 

3.24.5 length property
An artifact object MAY contain a property named length whose value is a non-negative integer specifying the length of the artifact in bytes.


---
In reply to: [276421529](https://github.com/Microsoft/sarif-sdk/pull/1428#discussion_r276421529) [](ancestors = 276421529,276421274)

Reply:
Sorry, i have reverted this change from this specific PR, because its a bit costly, we need to make fixes in some tests. I have this item in the excel sheet, lets do it in a separate PR

---
In reply to: [276427257](https://github.com/Microsoft/sarif-sdk/pull/1428#discussion_r276427257) [](ancestors = 276427257,276421529,276421274)